marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/ingestion/bulkIngester.sjs

```diff
@@ -29,11 +29,21 @@ const consts = require("/data-hub/5/impl/consts.sjs");
 const HubUtils = require("/data-hub/5/impl/hub-utils.sjs");
 const state = fn.head(xdmp.fromJSON(endpointState));
 
+const temporal = require("/MarkLogic/temporal.xqy");
+const temporalLib = require("/data-hub/5/temporal/hub-temporal.sjs");
+
 const work = fn.head(xdmp.fromJSON(workUnit));
 
 const uriPrefix = work.uriprefix != null ? work.uriprefix : "";
 
 const collections = work.collections != null ? work.collections.split(',') : [];
+const temporalCollections = temporalLib.getTemporalCollections().toArray().reduce((acc, col) => {
+    acc[col] = true;
+    return acc;
+}, {});
+
+let temporalCollection = collections.concat(collections).find((col) => temporalCollections[col]);
```

Comment:
Pinging @srinathgit  about this - there's some duplication between this and invoke-queue-write.sjs. I think what both clients are asking is - are any of these temporal collections, and if so, can you please give me back the array of collections that I should use for inserting the document? 

If the function returns null, then that means none of the collections are temporal, and the client should do an xdmp.documentInsert using whatever collections were provided. 

If the functions returns an array of length 1 or greater, then that means at least one of the collections was temporal, and the client should do a temporal.documentInsert. And the client doesn't have to worry about filtering out "latest" and "uri" either, the function will handle it. Which means the function should take "uri" as an argument too.

Reply:
@rjrudin ,
If I understand correctly, you are suggesting refactor the code so as to avoid duplication. If so, @SameeraPriyathamTadikonda , can you file a task and assign it to me, I will take it up once I am done with tasks for this sprint.

Reply:
A task works well, thanks!

Reply:
Created this task(https://project.marklogic.com/jira/browse/DHFPROD-6115) and assigned to @srinathgit .